tests/bsim/bluetooth/host/misc/sample_test/src/main.c

```diff
@@ -9,11 +9,33 @@
 #include "bs_tracing.h"
 #include "bstests.h"
 #include "babblekit/testcase.h"
+#include "testlib/log_utils.h"
 
 extern void entrypoint_dut(void);
 extern void entrypoint_peer(void);
 extern enum bst_result_t bst_result;
 
+unsigned long runtime_log_level = LOG_LEVEL_INF;
+
+static void test_args(int argc, char *argv[])
+{
+	size_t argn = 0;
+	const char *arg = argv[argn];
+
+	if (strcmp(arg, "log_level") == 0) {
+
+		runtime_log_level = strtoul(argv[++argn], NULL, 10);
+
+		if(runtime_log_level >= LOG_LEVEL_NONE && 
+			runtime_log_level <= LOG_LEVEL_DBG){
+			TEST_PRINT("Runtime log level configuration: %d", runtime_log_level);
+		} else {
+			TEST_FAIL("Invalid arguments to set log level: %d\n", runtime_log_level);
```

Comment:
```suggestion
			TEST_FAIL("Invalid arguments to set log level: %d", runtime_log_level);
```

Reply:
I have update peer module log registration to `LOG_MODULE_REGISTER(peer, LOG_LEVEL_INF)` now.

Reply:
@theob-pro thanks for the comments, updated the PR now :)

Reply:
Great, but regarding `peer.c` you just hard coded the log level, why not use the run time filtering like in the DUT?

Reply:
@theob-pro I just want to clarify if the same `log_level` is set to `peer` module as well.

Reply:
@theob-pro, When I pass runtime arg bigger than level the module is registered, I get the error. Is it expected? Below is the error:
```
d_00: WARNING: (src/bs_pc_base.c:146): Previous lock file found /tmp/bs_combine/tests_bsim_bluetooth_host_misc_sample_test/2G4.d0.lock
d_00: WARNING: (src/bs_pc_base.c:191): Found previous lock owned by DEAD process (pid was 962583), will attempt to take over
d_00: @00:00:00.000000 INFO: (WEST_TOPDIR/zephyr/tests/bsim/bluetooth/host/misc/sample_test/src/main.c:31): Runtime log level configuration: 4
d_01: @00:00:00.000000 INFO: (WEST_TOPDIR/zephyr/tests/bsim/bluetooth/host/misc/sample_test/src/main.c:31): Runtime log level configuration: 4
d_00: @00:00:00.000000  *** Booting Zephyr OS build v3.7.0-rc3-109-g82ac3a7eb0e1 ***
d_00: @00:00:00.000000 INFO: Test start: dut
d_01: @00:00:00.000000  *** Booting Zephyr OS build v3.7.0-rc3-109-g82ac3a7eb0e1 ***
d_01: @00:00:00.000000 INFO: Test start: peer
d_00: @00:00:00.000000  ASSERTION FAIL [result_level == new_level] @ WEST_TOPDIR/zephyr/tests/bluetooth/common/testlib/include/testlib/log_utils.h:22
d_01: @00:00:00.000000  ASSERTION FAIL [result_level == new_level] @ WEST_TOPDIR/zephyr/tests/bluetooth/common/testlib/include/testlib/log_utils.h:22
d_00: @00:00:00.000000  	3 4
d_01: @00:00:00.000000  	3 4
d_01: @00:00:00.000000  @ WEST_TOPDIR/zephyr/lib/os/assert.c:43
d_00: @00:00:00.000000  @ WEST_TOPDIR/zephyr/lib/os/assert.c:43
d_00: @00:00:00.000000  [00:00:00.000,000] <err> [main] os: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
d_01: @00:00:00.000000  [00:00:00.000,000] <err> [main] os: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
d_00: @00:00:00.000000  [00:00:00.000,000] <err> [main] os: Current thread: 0x825dbc0 (main)
d_00: @00:00:00.000000  [00:00:00.000,000] <err> [main] os: Halting system
d_01: @00:00:00.000000  [00:00:00.000,000] <err> [main] os: Current thread: 0x825dbc0 (main)
d_01: @00:00:00.000000  [00:00:00.000,000] <err> [main] os: Halting system
p_2G4: WARNING: (src/bs_pc_base.c:408): Device 0 left the party unsuspectingly.. I treat it as if it disconnected
timeout: the monitored command dumped core
/home/combine/zephyrproject/zephyr/tests/bsim/sh_common.source: line 22: 963019 Trace/breakpoint trap   $@
p_2G4: WARNING: (src/bs_pc_base.c:408): Device 1 left the party unsuspectingly.. I treat it as if it disconnected
timeout: the monitored command dumped core
/home/combine/zephyrproject/zephyr/tests/bsim/sh_common.source: line 22: 963020 Trace/breakpoint trap   $@
```
Is it a good idea to register `peer` and `dut` modules with `LOG_LEVEL_DBG`?

Reply:
My bad, the thing is you cannot set a level higher that the default of level of the module. So you were right to do `LOG_MODULE_REGISTER(..., LOG_LEVEL_DBG)` as this is the highest level available.

Reply:
@theob-pro thank for the confirmation. I have updated it now.

Reply:
Great, one last thing, you could remove the Kconfig now that it's not use anymore. (also remove the commented config in the project config)

Reply:
Hej @theob-pro , I have updated now

Reply:
Just remove `CONFIG_APP_LOG_LEVEL_DBG` in the comment, the others are still useful.